agent/util-scripts/pbench-register-tool

```diff
@@ -175,6 +175,16 @@ while true; do
 		shift
 		_do_test_labels=1
 		;;
+	--persistent)
+		shift
+		oldname=${name}
+		name=$(${pbench_bin}/util-scripts/get-internal-tool ${pbench_bin}/tool-scripts ${name} persistent)
+		;;
+	--transient)
```

Comment:
What happens if they don't specify `--persistent` or `--transient` for a "base" toolname that can be in either? Are we not handling that case at all? Should we default to one or the other, or complain that the tool class needs to be specified?

Also, what happens if someone specifies a persistent-only tool with --transient, or vice versa? Does the user get a meaningful and useful error message?

Reply:
By default, the tools "base name" represents the original tool type. Since the pcp tool is default persistent, simply doing a register for name=pcp will register the internal pcp tool (persistent). If pcp is registered with --persistent, it will check if "pcp" is a persistent tool already, and register it just the same. If it wasn't, then it would check for an internal "pcp-persistent", and register that internally. For pcp --transient it checks for a transient base tool called "pcp", fails to find it, then checks if there is a "pcp-transient". 

If mpstat was registered with nothing or --transient, it would be registered as "mpstat" internally, a default transient tool. If it was attempted with --persistent right now, it would fail to find a persistent "mpstat", then fail to find an internal "mpstat-persistent", and ultimately print a message saying "Could not find mpstat with type persistent... (insert rest of failure-to-register message + usage/tool options)". Eventually though (according to future plans), an "mpstat-persistent" will be made and then that option will work.